ui/text_displayer.js

```diff
@@ -227,11 +354,18 @@ shaka.ui.TextDisplayer = class {
 
     // Set white-space to 'pre-line' to enable showing line breaks in the text.
     captionsStyle.whiteSpace = 'pre-line';
-    captions.textContent = cue.payload;
     captionsStyle.backgroundColor = cue.backgroundColor;
     captionsStyle.color = cue.color;
     captionsStyle.direction = cue.direction;
 
+    // Add cue content
+    const content = this.buildCueElements_(cue.payload);
+    if (content) {
+      for (let i = 0; i < content.length; i++) {
```

Comment:
Here we would also prefer a for-of loop: for (const element of this.buildCueElements_(cue.payload))

Reply:
Per above, and resolved.